src/arith/mod.rs

```diff
@@ -336,4 +341,49 @@ arith_func!(BitAnd, bitand, af_bitand);
 arith_func!(BitOr, bitor, af_bitor);
 arith_func!(BitXor, bitxor, af_bitxor);
 arith_func!(Shl, shl, af_bitshiftl);
-arith_func!(Shr, shr, af_bitshiftr);
\ No newline at end of file
+arith_func!(Shr, shr, af_bitshiftr);
+
+macro_rules! arith_assign_func {
+    ($op_name:ident, $fn_name:ident, $func: ident) => (
+        impl $op_name<Array> for Array {
+
+            #[allow(unused_variables)]
+            fn $fn_name(&mut self, rhs: Array) {
+                let mut idxrs = Indexer::new();
+                idxrs.set_index(&Seq::<f32>::default(), 0, Some(false));
+                idxrs.set_index(&Seq::<f32>::default(), 1, Some(false));
+                let tmp = assign_gen(self as &Array, &idxrs,
+                                     & $func(self as &Array, &rhs, false));
+                mem::replace(self, tmp);
+            }
+        }
+    )
+}
+
+arith_assign_func!(AddAssign, add_assign, add);
+arith_assign_func!(SubAssign, sub_assign, sub);
+arith_assign_func!(MulAssign, mul_assign, mul);
+arith_assign_func!(DivAssign, div_assign, div);
+arith_assign_func!(RemAssign, rem_assign, rem);
+arith_assign_func!(ShlAssign, shl_assign, shiftl);
+arith_assign_func!(ShrAssign, shr_assign, shiftr);
+
+macro_rules! bit_assign_func {
+    ($op_name:ident, $fn_name:ident, $func: ident) => (
+        impl $op_name<Array> for Array {
+
+            #[allow(unused_variables)]
+            fn $fn_name(&mut self, rhs: Array) {
+                let mut idxrs = Indexer::new();
+                idxrs.set_index(&Seq::<f32>::default(), 0, Some(false));
+                idxrs.set_index(&Seq::<f32>::default(), 1, Some(false));
+                let tmp = assign_gen(self as &Array, &idxrs, & $func(self as &Array, &rhs));
+                mem::replace(self, tmp);
```

Comment:
@9prady9 : sorry for the delay. According to the docs `mem::replace` does **not** deinitialize the old value. It does however return the previous value. I think de-initializing manually here is the apt choice to prevent a dangling pointer. I'm not too worried about the actual array since that is a smart pointer in C++. Thoughts @pavanky ?